mcs/class/corlib/System/TypeSpec.cs

```diff
@@ -397,6 +504,45 @@ static TypeSpec Parse (string name, ref int p, bool is_recurse, bool allow_aqn)
 			p = pos;
 			return data;
 		}
+
+		// shallow copy of 'this', but without modifiers and without byref.
+		internal TypeSpec WithoutModifiers()
+		{
+			TypeSpec other = new TypeSpec ();
+			other.name = name;
+			other.assembly_name = assembly_name;
+			other.nested = nested;
+			other.generic_params = generic_params;
+
+			other.modifier_spec = null;
```

Comment:
null and false are default values, no need to set to them right after you called the constructor.